### PR TITLE
Make grading page functional

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

https://www.notion.so/uwblueprintexecs/Make-dummy-Test-Grading-platform-functional-4fdb0f5540e14c38acfed93e1d715013

## Implementation description

- allow admin to update story line status to "PENDING" for tests
- redirect to 404 on `/grade` if not a story test
- wrap grading buttons in tooltips to handle story test in TRANSLATE stage
- implement modal for fail user flow (not functional yet)
- modified StatusBadge to accomodate test grading statuses
- keep track of score state in `StoryTestGradingPage.tsx`

## Steps to test

1. `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data erase"`
2. `docker exec -it planet-read_py-backend_1 /bin/bash -c "python -m tools.insert_test_data"`
3. run "UPDATE story_translations SET stage='REVIEW' WHERE id=13;" in the db container
4. create story test using mutation below in Altair
5. change status to review using `UPDATE story_translations SET stage='REVIEW' WHERE id=14;`
6. login as admin
7. go to "Manage Tests" and click on "Grade"
8. verify that the action status are PENDING, CORRECT, PARTIALLY_CORRECT, and INCORRECT
9. play around with the status badges, verify that the score updates properly
10. click "Fail User", verify that the modal shows up (not functional yet)

#### Check review page
1. login as `planetread+dwightdeisenhower@uwblueprint.org`
2. click "View Translation" on "The Great Gatsby"
3. verify that the review platform works as normal, especially the action statuses

#### Create story test
```
mutation CreateStoryTest {
  createStoryTranslationTest(userId: 1, level:2, language:"FRENCH"){
    story{
      id
      translatorId
      storyId
      language
      stage
    }
  }
}
```

#### 

## What should reviewers focus on?

- does it work?
- is the styling okay?

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/StoryTestGradingPage.tsx [hunk 1]
```diff
@@ -47,25 +48,46 @@ const StoryTestGradingPage = () => {
...
+    const scoreReducer = (prev: number, cur: StoryLine) => {
+      const status = cur.status;
+      if (status === "Correct") {
+        return prev + 1;
+      } else if (status === "Partially Correct") {
+        return prev + 0.5;
+      }
+      return prev;
+    };
+    const newScore = translatedStoryLines.reduce(scoreReducer, 0);
```
Comment (gaoxk) [inline frontend/src/components/pages/StoryTestGradingPage.tsx:74]: niceeeeeeeee i luv reducers 😎

File: frontend/src/components/pages/StoryTestGradingPage.tsx [hunk 2]
```diff
@@ -97,6 +119,11 @@ const StoryTestGradingPage = () => {
     {
       fetchPolicy: "cache-and-network",
       onCompleted: (data) => {
+        console.log(data);
```
Comment (gaoxk) [inline frontend/src/components/pages/StoryTestGradingPage.tsx:122]: nit: remove consolelog

File: frontend/src/components/review/StatusBadge.tsx [hunk 3]
```diff
@@ -29,14 +31,39 @@ export type StatusBadgeProps = {
...
+  isTest: boolean;
 };
 
+type StatusOption = {
+  icon: IconType;
+  stage: string;
+  value: string;
+};
+
+const REVIEW_STATUS_OPTIONS: StatusOption[] = [
```
Comment (gaoxk) [inline frontend/src/components/review/StatusBadge.tsx:43]: noiceee i like the StatusOption lists

**Review by gaoxk — CHANGES_REQUESTED** [review]
image
everything worked when testing! only thing is that the padding is non existent now, mind keeping it in?

**Comment by Puepis** [comment]
> only thing is that the padding is non existent now, mind keeping it in?

oops mb will revert

**Review by gaoxk — CHANGES_REQUESTED** [review]
hmmm it looks good on the grading platform but padding is still missing from translation platform:
image
and review platform:
image

seems that the badges on the homepage story cards are really tight too now 
image
maybe the styling problem isnt from the ApproveAll component but the StatusBadges?

**Comment by Puepis** [comment]
i think styling should be fixed now! 🙏
grading
homepage
pending

**Review by gaoxk — APPROVED** [review]
NOICE

**Pull request merged**